File: APE/SW-AyLP-APE-02: Estructuras de Selección/Códigos_C++/numeroCuenta.java
import java.util.Scanner;

public class numeroCuenta {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        // En lugar de una clase/struct, usamos arreglos paralelos
        int[] numeroCuenta = new int[3];
        double[] saldo = new double[3];

        // Registro del cliente
        System.out.print("Ingrese DNI del cliente: ");
        String clienteDNI = scanner.next();

        // Inicializar hasta 3 cuentas (Índices 0, 1 y 2)
        for (int i = 0; i < 3; i++) {
            numeroCuenta[i] = i + 1; // Guarda el número de cuenta como 1, 2 y 3
            saldo[i] = 0.0;
        }

        int seleccionCuenta;
        int opcion = 0;
        double monto;

        // Menú principal
        do {
            System.out.print("\nSeleccione una cuenta (1-3): ");
            seleccionCuenta = scanner.nextInt();

            // Validar que la cuenta elegida sea correcta
            if (seleccionCuenta < 1 || seleccionCuenta > 3) {
                System.out.println("Cuenta no valida. Intente de nuevo.");
                continue;
            }

            // Convertimos la selección (1-3) al índice del arreglo (0-2)
            int index = seleccionCuenta - 1;

            System.out.println("Menu de opciones:");
            System.out.println("1. Ver atributos de la cuenta");
            System.out.println("2. Enviar dinero");
            System.out.println("3. Recibir dinero");
            System.out.println("4. Transferencia entre cuentas");
            System.out.println("5. Salir");
            System.out.print("Elija una opcion: ");
            opcion = scanner.nextInt();

            switch (opcion) {
                case 1:
                    System.out.println("Cuenta: " + numeroCuenta[index]);
                    System.out.println("Saldo: " + saldo[index]);
                    break;

                case 2:
                    System.out.print("Ingrese monto a enviar: ");
                    monto = scanner.nextDouble();
                    if (saldo[index] >= monto) {
                        saldo[index] -= monto;
                        System.out.println("Dinero enviado correctamente.");
                    } else {
                        System.out.println("Saldo insuficiente.");
                    }
                    break;

                case 3:
                    System.out.print("Ingrese monto a recibir: ");
                    monto = scanner.nextDouble();
                    saldo[index] += monto;
                    System.out.println("Dinero recibido correctamente.");
                    break;

                case 4:
                    System.out.print("Ingrese cuenta destino (1-3): ");
                    int destino = scanner.nextInt();

                    if (destino < 1 || destino > 3) {
                        System.out.println("Cuenta destino invalida.");
                        break;
                    }

                    int indexDestino = destino - 1;
                    System.out.print("Ingrese monto a transferir: ");
                    monto = scanner.nextDouble();

                    if (saldo[index] >= monto) {
                        saldo[index] -= monto;
                        saldo[indexDestino] += monto;
                        System.out.println("Transferencia realizada.");
                    } else {
                        System.out.println("Saldo insuficiente.");
                    }
                    break;

                case 5:
                    System.out.println("Saliendo del sistema...");
                    break;

                default:
                    System.out.println("Opcion invalida.");
                    break;
            }
        } while (opcion != 5);

        scanner.close();
    }
}

File: APE/SW-AyLP-APE-02: Estructuras de Selección/Códigos_C++/numeroCuenta.cpp
#include <iostream>
#include <string>

using namespace std;

// Representación del 'Registro' de PSeInt en C++
struct Cuenta {
    int numeroCuenta;
    double saldo;
};

int main() {
    string clienteDNI;
    
    // Crear un arreglo para guardar 3 estructuras de tipo Cuenta
    Cuenta cuentas[3];

    // Registro del cliente
    cout << "Ingrese DNI del cliente: ";
    cin >> clienteDNI;

    // Crear e inicializar hasta 3 cuentas (Índices 0, 1 y 2)
    for (int i = 0; i < 3; i++) {
        cuentas[i].numeroCuenta = i + 1; // Guarda el número de cuenta como 1, 2 y 3
        cuentas[i].saldo = 0.0;
    }

    int seleccionCuenta;
    int opcion;
    double monto;
    int destino;
    int indexDestino;

    // Menú principal
    do {
        cout << "\nSeleccione una cuenta (1-3): ";
        cin >> seleccionCuenta;

        // Validar que la cuenta elegida sea correcta antes de continuar
        if (seleccionCuenta < 1 || seleccionCuenta > 3) {
            cout << "Cuenta no valida. Intente de nuevo." << endl;
            continue;
        }

        // Convertimos la selección (1-3) al índice del arreglo (0-2)
        int index = seleccionCuenta - 1;

        cout << "Menu de opciones:" << endl;
        cout << "1. Ver atributos de la cuenta" << endl;
        cout << "2. Enviar dinero" << endl;
        cout << "3. Recibir dinero" << endl;
        cout << "4. Transferencia entre cuentas" << endl;
        cout << "5. Salir" << endl;
        cout << "Elija una opcion: ";
        cin >> opcion;

        switch (opcion) {
            case 1:
                cout << "Cuenta: " << cuentas[index].numeroCuenta << endl;
                cout << "Saldo: " << cuentas[index].saldo << endl;
                break;

            case 2:
                cout << "Ingrese monto a enviar: ";
                cin >> monto;
                if (cuentas[index].saldo >= monto) {
                    cuentas[index].saldo -= monto;
                    cout << "Dinero enviado correctamente." << endl;
                } else {
                    cout << "Saldo insuficiente." << endl;
                }
                break;

            case 3:
                cout << "Ingrese monto a recibir: ";
                cin >> monto;
                cuentas[index].saldo += monto;
                cout << "Dinero recibido correctamente." << endl;
                break;

            case 4:
                cout << "Ingrese cuenta destino (1-3): ";
                destino;
                cin >> destino;

                if (destino < 1 || destino > 3) {
                    cout << "Cuenta destino invalida." << endl;
                    break;
                }

                indexDestino = destino - 1;
                cout << "Ingrese monto a transferir: ";
                cin >> monto;

                if (cuentas[index].saldo >= monto) {
                    cuentas[index].saldo -= monto;
                    cuentas[indexDestino].saldo += monto;
                    cout << "Transferencia realizada." << endl;
                } else {
                    cout << "Saldo insuficiente." << endl;
                }
                break;

            case 5:
                cout << "Saliendo del sistema..." << endl;
                break;

            default:
                cout << "Opcion invalida." << endl;
                break;
        }
    } while (opcion != 5);

    return 0;
}
